script.lua

```diff
@@ -3639,9 +3647,18 @@ function runSpiritSetup()
     for color,data in pairs(selectedColors) do
         if data.zone then
             for _,obj in pairs(data.zone.getObjects()) do
-                if obj.hasTag("Setup") and not obj.getVar("setupComplete") then
-                    local success = obj.call("doSetup", {color=color})
-                    obj.setVar("setupComplete", success)
+                if obj.hasTag("Setup") then
+                    -- Whether the object has already done setup is stored in its script state, to persist across save/reload
+                    local json = JSON.decode(obj.script_state)
+                    if not json then
+                        json = {}
+                    end
+                    if not json.setupComplete then
+                        local success = obj.call("doSetup", {color=color})
+                        json.setupComplete = success
+                        obj.script_state = JSON.encode(json)
+                        break
```

Comment:
i'm not seeing this break exist before the change, so i think adding it would be a bug. in theory a spirit could have multiple objects in their play area that need setup

Reply:
Ah, you're absolutely right. That's a result of copying the code over from `onObjectEnterScriptingZone()`. It feels like there's a fair bit of code duplication in the parts of these two functions that run `doSetup` on objects, but they're different enough that I couldn't find an elegant way of removing the duplication. I'll just knock out the break for now; that should fix it.